### DISCUSSION TITLE
Use html string in Head

### DISCUSSION THREAD

**Discussion by andriiradkevych** [freshframework/fresh#2360] [General]
```ts
import { Head } from '$fresh/runtime.ts';

interface Props {
  yoast_head: string;
}

const Seo = ({ yoast_head }: Props) => {
  return (
    <Head>
      {yoast_head}
    </Head>
  );
};
```

I have `Seo` component in which I need to parse `yoast_head` html string . Example of `yoast_head` is:

```
const yoast_head = '<title>Title</title><meta></meta>`
```

How can I transform it to jsx , or maybe there is some method in api , which I can use ?

I see that there exist plugins https://fresh.deno.dev/docs/concepts/plugins#creating-a-plugin , by which I can change body , but it isn't  clear how to change `head` from there

**Comment by CAYdenberg** [comment]
Looks like the API for writing to the `head` will change soon: https://github.com/denoland/fresh/issues/2363 (scroll down to "Adding head elements from handlers")

Either way in your case I would consider parsing using [Deno-dom](https://docs.deno.com/runtime/manual/advanced/jsx_dom/deno_dom) and then writing pre-defined tags, rather than spitting out html from Yoast (don't you need to rewrite the root url, anyway?)

You _might_ be able to write using dangerouslySetInnerHtml, but I'm not sure quite what that would look like.

**Reply by andriiradkevych** [reply]
Could you pls attach some example with `<Head>` ,to make sure that we on same page ?

**Reply by CAYdenberg** [reply]
Something like (based on your example)

```js
const doc = new DOMParser().parseFromString(text, "text/html");
const title = doc.querySelector('title').innerText;
const meta = Array.from(doc.querySelectorAll('meta'));

return (
  <Head>
    <title>{title}</title>
    {meta.map((el, i) => 
      <meta key={i} name={el.getAttribute('name')} content={el.getAttribute('content')} />
    )}
  </Head>
);
```

But I would also carefully inspect the data by hand - console.log the `doc` and see what you need to include, what you need to modify, and what you can discard.

**Reply by marvinhagemeister** [reply]
Would be cool to add support for setting `dangerouslySetInnerHTML` on fragments in Preact.

**Reply by andriiradkevych** [reply]
@marvinhagemeister  as I see , discussion in preact community will not be proceeded , could you pls try to make some plugin opportunity to have access to `head` where I will be able to change `head` string ( same as `render` )  ?

**Reply by marvinhagemeister** [reply]
Exposing `head` is not really straightforward due to the way Fresh 1.x does rendering internally as it does a few passes to inject custom tags. The proper solution is to extend `Fragments` to support `dangerouslySetInnerHTML`. I currently have my hands full with shipping Fresh 2.0 and I don't have the bandwidth to look at this at the moment.

In the meantime you should be able to intercept any responses in a middleware and mess with it in any way you desire.
